### PR TITLE
use deprecated dispatcher with IEventListener

### PR DESCRIPTION
**Pull request by ArtificialOwl**

### CODE REVIEW

**Comment by ArtificialOwl** [comment]
fix 

```
App globalsiteselector threw an error during app.php load: Argument 2 passed to OC\\EventDispatcher\\EventDispatcher::addListener() must be callable, string given, called in /home/maxence/sites/globalscale/master/nextcloud/apps/globalsiteselector/lib/AppInfo/Application.php on line 75
```

**Review by CarlSchwan — APPROVED** [review]

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -70,9 +71,19 @@ private function registerMasterHooks(IAppContainer $c) {
...
+			/** @var IEventDispatcher $eventDispatcher */
+			$eventDispatcher = $c->getServer()->query(IEventDispatcher::class);
+			$eventDispatcher->addListener(
+				AddContentSecurityPolicyEvent::class,
+				function (AddContentSecurityPolicyEvent $event) {
+					/** @var AddContentSecurityPolicyListener $listener */
+					$listener = \OC::$server->get(AddContentSecurityPolicyListener::class);
+					$listener->handle($event);
+				}
+			);
```
Comment (juliushaertl) [inline lib/AppInfo/Application.php:84]: Right, I think I mixed that up with the method on IRegistrationContext

Maybe it should just use addServiceListener then:

```suggestion
			$eventDispatcher->addServiceListener(AddContentSecurityPolicyEvent::class, AddContentSecurityPolicyListener::class);
```

Comment (ArtificialOwl) [reply lib/AppInfo/Application.php:84]: yes, should work too. Haven't tested, but we need to choose one at least :) 

Create your PR if you want to implement your solution, i'll test & vouch

Comment (juliushaertl) [reply lib/AppInfo/Application.php:84]: Sure, pushed to https://github.com/nextcloud/globalsiteselector/pull/52 :) testing would be very welcome

**Pull request closed without merging**